src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheckTest.java

```diff
@@ -741,16 +737,46 @@ public void testNoPackage2() throws Exception {
 
         createChecker(checkConfig);
         final String[] expected = {
-            "3: " + getCheckMessage(MSG_LINE_SEPARATOR,
-                "com.puppycrawl.tools.checkstyle.utils.AnnotationUtil.containsAnnotation"),
             "7: " + getCheckMessage(MSG_LINE_SEPARATOR,
                 "com.sun.accessibility.internal.resources.*"),
-            "11: " + getCheckMessage(MSG_LINE_SEPARATOR, "java.util.Arrays"),
-            "19: " + getCheckMessage(MSG_LINE_SEPARATOR,
-                "org.apache.commons.beanutils.converters.ArrayConverter"),
         };
         verify(checkConfig, getNonCompilablePath("InputCustomImportOrderNoPackage2.java"),
             expected);
     }
 
+    @Test
+    public void testInputCustomImportOrderMultipleSeparators() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(CustomImportOrderCheck.class);
+        checkConfig.addAttribute("customImportOrderRules",
+                "STATIC###STANDARD_JAVA_PACKAGE###THIRD_PARTY_PACKAGE");
+        checkConfig.addAttribute("separateLineBetweenGroups", "true");
+
+        createChecker(checkConfig);
+        final String[] expected = {
+            "7: " + getCheckMessage(MSG_LINE_SEPARATOR,
+                "java.util.Map"),
+            "13: " + getCheckMessage(MSG_LINE_SEPARATOR,
+                "com.google.common.annotations.Beta"),
+        };
+        verify(checkConfig, getNonCompilablePath("InputCustomImportOrderMultipleSeparators.java"),
+            expected);
+    }
+
+    @Test
+    public void testGetCountOfEmptyLinesBefore() throws Exception {
+        final Class<?> clazz = CustomImportOrderCheck.class;
+        final Method method = clazz.getDeclaredMethod("getCountOfEmptyLinesBefore",
```

Comment:
This is not ok.
Pure unit test methods (that test particular method in class) are exceptions and you need to explain in comment above test method why it is not possible to cover code in regular way - by Inputs.
All should be covered by Inputs and "whole checkstyle" execution.
Please explain in details why you can not cover code by Input files.

Reply:
There's an edge case that, it is not possible to pass 100% branch coverage by inputs. The `CustomImportOrder` checks empty lines between import groups, and the current logic, only input having multiple import groups like this typical one,
```java
import static java.awt.Button.ABORT;

import java.util.Map; // check empty lines between static and java group
import java.util.Map.Entry;
import java.util.NoSuchElementException;
....
``` 
will pass into `CustomImportOrder#getCountOfEmptyLinesBefore(int lineNo, final String... lines)` with parameter `lineNo = 3`,  and the `lineBeforeIndex ` will never reach `0` because short cut by `CommonUtil.isBlank(lines[lineBeforeIndex])`  , thus not reaching 100% branch coverage.

Reply:
Comments added :)

Reply:
![image](https://user-images.githubusercontent.com/812984/64450921-f032a200-d097-11e9-9376-cb8540b069e9.png)

Reply:
I reverted all your changes in getCountOfEmptyLinesBefore and all tests pass (I commented out testGetCountOfEmptyLinesBefore too) ... weird.
as non of test require them - please revert them.

Such line was covered before your changes, your changes are not affecting test, so it mean very suspicious in your code, lets remove all unnecessary changes.

Reply:
It is expected that the branch coverage does not reach 100%  if you comment out `testGetCountOfEmptyLinesBefore()`, see the jacoco report of CustomImportOrderCheck line number 767 you posted above. 

For the changes in getCountOfEmptyLinesBefore,
```java
while (lineBeforeIndex > -1
```
has no difference than
```java
while (lineBeforeIndex >= 0
```
, the former one pass mutation test but the latter does not.

Reply:
Branches coverage of `getCountOfEmptyLinesBefore(int, String[])` decrease to 75% after comment out `testGetCountOfEmptyLinesBefore()` .
![image](https://user-images.githubusercontent.com/3699228/64475825-3718b880-d1ba-11e9-9ec4-6c8094ad6722.png)

Reply:
I revert the change `while (lineBeforeIndex > -1` in `getCountOfEmptyLinesBefore`, let's see if it would pass mutation test pitest-imports.

Reply:
Wouldn't you be able to pass coverage without the reflection test if you just had an input file where everything was on a single line? Just put the package, multiple imports, and class all on a single line.

Reply:
I've never coding like that, sounds interesting 😄 , let me have a try.

Reply:
Updated as you suggested. Avoiding pure unit test. More input added to reach 100% branches coverage.

Reply:
I tried again. It's weird that after reverting `while (lineBeforeIndex > -1` to original `while (lineBeforeIndex >= 0` in `getCountOfEmptyLinesBefore(int)`, the mutation test would not pass, but the master branch(without this PR) does not have such issue, just as @romani figured out.

I'm not familiar with mutation test, but how does it behavior like this? It would be great if someone could explain in more detail.

Reply:
I looked briefly at the mutation earlier.
Mutation testing is basically changing your code in memory, running it through the test suite, and looking for a test to fail. It proves that the code you wrote is needed and atleast 1 test case shows that it should stay as it is.
The mutation surviving for this while loop is http://pitest.org/quickstart/mutators/#CONDITIONALS_BOUNDARY . It mutates `>=` and changes it to `>`.
You can make this change your self and debug the code to understand why there are no test failures. It could help you understand better why the code seems like it has to change.

Reply:
The changes in `testNoPackage2()` decrease code coverage, and thus the mutation test failed. It looks like need more inputs to prevent mutation test failure.